src/main/kotlin/no/fdk/dataservicecatalog/handler/RDFHandler.kt

```diff
@@ -1,21 +1,279 @@
+import org.apache.jena.rdf.model.ResourceFactory
 import org.apache.jena.riot.Lang
+import org.apache.jena.sparql.vocabulary.FOAF
+import org.apache.jena.util.FileUtils
+import org.apache.jena.util.URIref
+import org.apache.jena.vocabulary.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Component
+import java.io.StringWriter
 
 @Component
-class RDFHandler(private val repository: DataServiceRepository) {
+class RDFHandler(private val repository: DataServiceRepository, private val properties: ApplicationProperties) {
 
     fun findAll(lang: Lang): String {
-        return ""
+        val model = createModel()
+
+        repository.findAllByStatus(Status.PUBLISHED)
+            .groupBy(DataService::catalogId)
+            .forEach { (catalogId, dataServices) ->
+                catalogId?.let {
+                    model.addCatalog(it, getCatalogUri(), getOrganizationUri(), getPublisherUri())
+                }
+
+                dataServices.forEach { dataService ->
+                    model.addDataServiceToCatalog(dataService, getCatalogUri(), getDataServiceUri())
+                    model.addDataService(dataService, getDataServiceUri())
+                }
+            }
+
+        return model.serialize(lang)
     }
 
     fun findById(catalogId: String, lang: Lang): String {
-        return ""
+        val model = createModel()
+
+        repository.findAllByCatalogIdAndStatus(catalogId, Status.PUBLISHED)
+            .forEach { dataService ->
+                dataService.catalogId?.let {
+                    model.addCatalog(it, getCatalogUri(), getOrganizationUri(), getPublisherUri())
+                }
+
+                model.addDataServiceToCatalog(dataService, getCatalogUri(), getDataServiceUri())
+                model.addDataService(dataService, getDataServiceUri())
+            }
+
+        return model.serialize(lang)
+    }
+
+    fun findDataServiceById(catalogId: String, dataServiceId: String, lang: Lang): String {
+        val dataService = repository.findDataServiceById(dataServiceId)
+            ?.takeIf { it.catalogId == catalogId }
+            ?: throw NotFoundException("Data Service with id: $dataServiceId not found in Catalog with id: $catalogId")
+
+        val model = createModel()
+        model.addDataService(dataService, getDataServiceUri())
+
+        return model.serialize(lang)
+    }
+
+    private fun createModel(): Model {
+        return ModelFactory.createDefaultModel().apply {
+            setNsPrefixes(
+                mapOf(
+                    "dcat" to DCAT.NS, "dct" to DCTerms.NS, "rdf" to RDF.uri, "vcard" to VCARD4.NS, "foaf" to FOAF.NS
+                )
+            )
+        }
+    }
+
+    private fun getCatalogUri(): String {
+        return buildUri(properties.baseUri, "/catalogs/")
+    }
+
+    private fun getDataServiceUri(): String {
+        return buildUri(properties.baseUri, "/data-services/")
```

Comment:
Denne skal peke til rdf-endepunktet til datatjenesten
```suggestion
    private fun getDataServiceUri(catalogId: String): String {
        return buildUri(properties.baseUri, "/catalogs/$catalogId/data-services/")
```